qiskit/circuit/quantumcircuit.py

```diff
@@ -107,7 +108,7 @@
 VALID_QASM2_IDENTIFIER = re.compile("[a-z][a-zA-Z_0-9]*")
 
 
-class QuantumCircuit:
+class QuantumCircuit(Operation):
```

Comment:
Currently a circuit is not an `Operation` according to the definition that an `Operation` is what you can attach to a circuit, right? If we currently add a circuit to another circuit it either converts it to an `Instruction` (when using `append`) or copy+pastes the `Instruction`s (when using `compose`). Either way, a circuit itself is right now never part of another circuit, is it?

Reply:
Since being an `Operation` just indicates he has the properties; `name`, `num_qbits`, `params`, etc., `QuantumCircuit` does have this so seems currently appropriate. Perhaps this approach could help simplify going between circuits and instructions. However, perhaps since this is not clearly defined in this project we can put it off for a later pr. If removing this here we can also remove the call to `super` below. Doing this seems to not break any tests so it doesn't seem required for the pr.

Reply:
Ah I see, if that's the definition then the circuit would qualify, yes 🙂 Then I'm only concerned about `params` and `num_params` methods along with the already existing `parameters` and `num_parameters`, that would be quite confusing.

Reply:
Removed `Operation` mixin from `QuantumCircuit` - to be handled in a future PR